include/rocksdb/perf_context.h

```diff
@@ -169,6 +189,8 @@ struct PerfContext {
   uint64_t env_lock_file_nanos;
   uint64_t env_unlock_file_nanos;
   uint64_t env_new_logger_nanos;
+  std::map<uint32_t, PerfContextByLevel*>* level_to_perf_context;
```

Comment:
You chose `PerfContextByLevel*` instead of `PerfContextByLevel`. What's the reason behind it? Is it to save memory waste? One clear downside is that, there will be memory leak after a thread is terminated.